### PR TITLE
marktext: fix nix-env version parsing

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by nh2** [comment]
Apparently other packages like `firefox` and `julia` put the `-bin` into the `pname`, e.g. here:

https://github.com/NixOS/nixpkgs/blob/f678ad7c7a306f5a073fbb68e93a074ca2d1f054/pkgs/development/compilers/julia/1.6-bin.nix#L4-L5

Should we do that as well instead?

That way e.g. `nix search` makes clear that this is a binary package.

**Comment by SuperSandro2000** [comment]
It makes sense if there are multiple packages and ideally it should be part of meta because parsing pname for information is always a bad idea.

File: pkgs/applications/misc/marktext/default.nix [hunk 1]
```diff
@@ -35,7 +34,7 @@ appimageTools.wrapType2 rec {
 
   extraInstallCommands = ''
     # Strip version from binary name.
-    mv $out/bin/${name} $out/bin/${pname}
+    mv $out/bin/${pname}-${version}-binary $out/bin/${pname}
```
Comment (nh2) [inline pkgs/applications/misc/marktext/default.nix:37]:
```suggestion
    mv $out/bin/${pname}-${version} $out/bin/${pname}
```

**Review by nh2 — CHANGES_REQUESTED** [review]
The file doesn't end with `-binary` in this case.

Otherwise, makes sense to me, good to merge after fixing this.

**Review by nh2 — APPROVED** [review]

**Pull request merged**